### DISCUSSION TITLE
allocating SPI failed: Operation not supported (95)

### DISCUSSION THREAD

**Discussion by michaelgiordano** [strongswan/strongswan#2505] [Q&A]
```
~$ sudo ipsec restart && sudo service xl2tpd restart && sudo ipsec up myvpn
Stopping strongSwan IPsec...
Starting strongSwan 5.9.1 IPsec [starter]...
initiating Main Mode IKE_SA myvpn[1] to 76.XXX.XX.X
generating ID_PROT request 0 [ SA V V V V V ]
sending packet: from 192.168.1.116[500] to 76.XXX.XX.X[500] (240 bytes)
received packet: from 76.XXX.XX.X[500] to 192.168.1.116[500] (160 bytes)
parsed ID_PROT response 0 [ SA V V V V ]
received XAuth vendor ID
received DPD vendor ID
received FRAGMENTATION vendor ID
received NAT-T (RFC 3947) vendor ID
selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
sending packet: from 192.168.1.116[500] to 76.XXX.XX.X[500] (244 bytes)
received packet: from 76.XXX.XX.X[500] to 192.168.1.116[500] (244 bytes)
parsed ID_PROT response 0 [ KE No NAT-D NAT-D ]
local host is behind NAT, sending keep alives
remote host is behind NAT
generating ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
sending packet: from 192.168.1.116[4500] to 76.XXX.XX.X[4500] (108 bytes)
received packet: from 76.XXX.XX.X[4500] to 192.168.1.116[4500] (76 bytes)
parsed ID_PROT response 0 [ ID HASH ]
IKE_SA myvpn[1] established between 192.168.1.116[192.168.1.116]...76.XXX.XX.X[192.168.1.109]
scheduling reauthentication in 10055s
maximum IKE_SA lifetime 10595s
allocating SPI failed: Operation not supported (95)
unable to get SPI
allocating SPI from kernel failed
establishing connection 'myvpn' failed
```

```
~$ uname -a
Linux raspberrypi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
```

My /etc/ipsec.conf looks like
```
config setup
  charondebug="ike 2, knl 2, cfg 2, net 2"

conn myvpn
  auto=add
  keyexchange=ikev1
  authby=secret
  type=transport
  left=%defaultroute
  leftprotoport=17/1701
  rightprotoport=17/1701
  right=X.X.X
  rightid=192.168.1.109
  ike=aes128-sha1-modp1024
  esp=aes128-sha1
```

I am trying to setup a strongswan client on a raspberry pi to access an l2tp VPN. I was able to get it talking to my VPN server successfully, but then I get this SPI error which is failing the connection. Any ideas? I have updated to the latest stable kernel, and checked for some relevant modules like ah4, esp4, xfrm_user, xfrm_algo, xfrm_tunnel4, etc. I've also tried switching to 32bit arch, which didn't work. Thanks for the help.

**Comment by tobiasbrunner** [comment]
Apparently there is still something wrong with your kernel.

> I've also tried switching to 32bit arch, which didn't work.

What do you mean? Generally userland and kernel have to match (there a translation subsystem in newer kernels for XFRM, but not sure if that's enabled by default).

**Reply by michaelgiordano** [reply]
I mean that I tried booting with `arm_64bit=0`
in the raspberry pi's `/boot/config.txt` per the discussion on [https://forums.raspberrypi.com/viewtopic.php?t=344246&start=350#p2102493](https://forums.raspberrypi.com/viewtopic.php?t=344246&start=350#p2102493)

Any idea what specifically is wrong with my kernel? Is there some way for strongswan to log more information about the SPI error?

**Reply by tobiasbrunner** [reply]
> I mean that I tried booting with `arm_64bit=0`

Hm, that doesn't really make sense to me (unless perhaps if the userland is 32-bit and just the kernel was switched from 32-bit to 64-bit with an update).

> Any idea what specifically is wrong with my kernel?

No, sorry.

>  Is there some way for strongswan to log more information about the SPI error?

It's a kernel issue, so all the information strongSwan has are already logged. You can check the kernel logs, but I doubt you'll find anything related to this.

**Reply by michaelgiordano** [reply]
Ok, thanks for the help.
